native_client/client.cc

```diff
@@ -238,6 +255,65 @@ ProcessFile(ModelState* context, const char* path, bool show_times)
   }
 }
 
+std::vector<meta_word>
+WordsFromMetadata(Metadata* metadata)
+{
+  std::vector<meta_word> word_list;
+
+  std::string word = "";
+  float word_start_time = 0;
+
+  // Loop through each character
+  for (int i = 0; i < metadata->num_items; i++) {
+    MetadataItem item = metadata->items[i];
+
+    if (strcmp(item.character, " ") != 0) {
+      word.append(item.character);
+    }
+
+    // Word boundary is either a space or the last character in the array
+    if (strcmp(item.character, " ") == 0 
```

Comment:
Again the same question here as above...

This is different from the original code. The original code used [isspace()](https://en.cppreference.com/w/cpp/string/byte/isspace) which would include more varied types of whitespace characters. This includes only " "

Also, the idea of splitting on the [25 types](https://en.wikipedia.org/wiki/Whitespace_character#Unicode) of UTF-8 whitespaces seems to be ignored. Are you planning to do this?

Reply:
`isspace` just checks for tabs, newlines, etc, which are (should be!) irrelevant for DeepSpeech. We already depend on having a single space label in the decoder here: https://github.com/mozilla/DeepSpeech/blob/master/native_client/ctcdecode/ctc_beam_search_decoder.cpp#L32

Checking for all 25 types of UTF-8 whitespaces won't add anything to this code, it still won't work with Chinese/Japanese, where there's no space characters between words, and there's other explicit checks for ASCII space elsewhere.

Reply:
> (CJK)...where there's no space characters between words...

CJK has [ideographic spaces](https://en.wiktionary.org/wiki/ideographic_space) which are not used all of the time, but are used.

Reply:
Here's a scenario where handling ideographic spaces is relevant in this code:

1. We have a working Chinese model
2. It is trained on enough data containing ideographic spaces to be able to correctly predict them
3. We've come up with a solution for [splitting Chinese text into words in general](https://stackoverflow.com/a/49935752/346048)
4. That general solution fails to work with ideographic spaces
5. We can't just replace ideographic spaces with normal spaces

Doesn't seem very relevant to this PR...

Reply:
This code isn't just for us.

Reply:
My point is that this code is not for using with Chinese/Japanese text, and no amount of space checking will make it work with Chinese/Japanese text, so why bother.

Reply:
If that's the case, what breaks for CJK? Or do you generally mean that spaces aren't used?

Also the change I'm suggesting is trivial.
```C++
if (strcmp(item.character, " ") == 0 
```
would change to
```C++
if (strcmp(item.character, " ") == 0 || strcmp(item.character, u8"　") == 0
```
the addition is an ideographic space.

Reply:
The change I was objecting to was the original suggestion of having 25 conditions on this if for every Unicode whitespace character. I mean that spaces aren't generally used, ideographic or not, so splitting on words is a much more complicated problem than this code handles. I will defer to your judgement on this so as to not waste any more of our times, but I still think adding the check here is unneeded and misleading, as it may indicate to users that this code is meant to be used with Chinese or Japanese and was designed/tested for that.

Reply:
I agree splitting on words is super hard to do for all languages and it would be impossible to do correctly here without including some massive library. However, I was hoping to get a MVP here for non-Latinate languages.

But, as the next language we are really concentrating on is simplified Chinese Mandarin maybe we can defer until we have the simplified Chinese Mandarin data set, then fix this however it needs to be fixed?

Reply:
Sounds like a good plan to me.